File: A20/A20/Task.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace A20
{
    public class Task
    {
        String title;
        String description;
        DateTime dueDate;
        DateTime creationDate;
        State taskState;
        bool isArchived;

        public String Title
        {
            get { return title; }
            set { title = value; }
        }
        public String Desc
        {
            get { return description; }
            set { description = value; }
        }
        public DateTime Due
        {
            get { return dueDate; }
            set { dueDate = value; }
        }
        public DateTime Creation
        {
            get { return creationDate; }
            set { creationDate = value; }
        }
        public State TaskState
        {
            get { return taskState; }
            set { taskState = value; }
        }
        public Task(String name, String desc, DateTime due, DateTime creation, State TState)
        {
            title = name;
            description = desc;
            dueDate = due;
            creationDate = creation;
            taskState = TState;

        }
        public bool Archived
        {
            set { isArchived = value; }
            get { return isArchived; }
        }

        public enum State
        {
            Todo,
            InProgress,
            Done,
            Archived
        }
    }
}

File: java/A20/src/ch/ictbz/todolist/Task.java
package ch.ictbz.todolist;

import java.time.LocalDate;

public class Task {
    private String title;
    private String description;
    private LocalDate dueDate;
    private LocalDate creationDate;
    private TaskState state;
    private boolean isArchived;


    public Task(String Title, String Desc, LocalDate Due, LocalDate Creation, TaskState State) {
        title = Title;
        description = Desc;
        dueDate = Due;
        creationDate = Creation;
        state = State;
    }

    public enum TaskState {
        Todo,
        InProgress,
        Done,
        Archived
    }

    public String getTitle() {
        return title;
    }
    public void setTitle(String value) {
        title = value;
    }

    public String getDescription() {
        return description;
    }
    public void setDescription(String value) {
        description = value;
    }

    public LocalDate getDueDate() {
        return dueDate;
    }
    public void setDueDate(LocalDate value) {
        dueDate = value;
    }

    public LocalDate getCreationDate() {
        return creationDate;
    }

    public TaskState getState() {
        return state;
    }
    public void setState(TaskState value) {
        state = value;
    }
}
